consensus/src/pipeline/execution_schedule_phase.rs

```diff
@@ -92,15 +92,22 @@ impl StatelessPipeline for ExecutionSchedulePhase {
         //      ExecutionWait phase is never kicked off.
         let fut = tokio::task::spawn(async move {
             let mut results = vec![];
-            for (block, fut) in itertools::zip_eq(ordered_blocks, futs) {
+            // wait for all futs so that lifetime_guard is guaranteed to be dropped only
+            // after all executor calls are over
+            for (block, fut) in itertools::zip_eq(&ordered_blocks, futs) {
```

Comment:
How about put these `futs` in a `FuturesOrdered` and poll them all together?

Reply:
doesn't matter, they are already spawned and running in parallel?

Reply:
They run in parallel, but I notice there is a future [here](https://github.com/aptos-labs/aptos-core/blob/e6226eb030f681c74bb518a297c8b05933dbc9b2/consensus/src/state_computer.rs#L228-L267) where we do some post-processing with some `.await`. And, possibility that in the future, this future can have other things. 

On the other hand, polling one by one would keep the logs sequential. Otherwise, we risk reading unordered logs when debugging. Let's leave it as-is then.

Reply:
Also not sure if things in the post processing wants to be sequential.. For example, does the mempool tolerate seeing the notifications out of order?